Stratis.Bitcoin.Features.Consensus/CoinViews/CachedCoinView.cs

```diff
@@ -228,9 +236,23 @@ private CachedCoinView(ILoggerFactory loggerFactory, StakeChainStore stakeChainS
         /// <summary>
         /// Finds all changed records in the cache and persists them to the underlaying coinview.
         /// </summary>
-        public async Task FlushAsync()
+        /// <param name="force"><c>true</c> to enforce flush, <c>false</c> to flush only if <see cref="lastCacheFlushTime"/> is older than <see cref="CacheFlushTimeIntervalSeconds"/>.</param>
+        /// <remarks>
+        /// WARNING: This method can only be run from <see cref="ConsensusLoop.Execute(System.Threading.CancellationToken)"/> thread context
+        /// or when consensus loop is stopped. Otherwise, there is a risk of race condition when the consensus loop accepts new block.
+        /// </remarks>
+        public async Task FlushAsync(bool force = true)
         {
-            this.logger.LogTrace("()");
+            this.logger.LogTrace("({0}:{1})", nameof(force), force);
+
+            DateTime now = this.dateTimeProvider.GetUtcNow();
+            if (!force && ((now - this.lastCacheFlushTime).TotalSeconds < CacheFlushTimeIntervalSeconds))
```

Comment:
The only issue here is that `CacheMaxItemsDefault` is not applied, so the cache might grow beyond that, probably if we flash every 60 sec it wont.

Perhaps we add that as a condition as well or delete it all together.

Reply:
I think it's ok as we moved from a size-based threshold to a time-based one.
Checking both seems superfluous to me as like you said, flushing every 60 sec should be enough.

Reply:
Actually, this is something else in my understanding. We use that max items limit for calling Evict inside that FCA method. That actually removes some items from the cache, making the memory consumption of the structure lower. 

However, flushing is not about making the memory consumption lower - note that we don't call Evict in this case. Flushing is only to persist unsaved data to disk, that is all. It won't make the structure smaller. It is true, that flushing allows the Evict to get rid of more items, but those are two separated things. The max items limit is still used for eviction, but this is now separated from flushing.

Reply:
Ah right yes correct flush and evict two different things, all good then ignore my comment.